app/javascript/components/modals/save-aoi/actions.js

```diff
@@ -0,0 +1,166 @@
+import { createAction, createThunkAction } from 'redux-tools';
+import { setComponentStateToUrl } from 'utils/stateToUrl';
+import { setAreasProvider, deleteAreaProvider } from 'services/areas';
+import {
+  setArea,
+  setAreas,
+  setActiveArea
+} from 'providers/areas-provider/actions';
+import { setDrawnGeostore } from 'pages/map/actions';
+
+import { MAP } from 'router';
+
+export const setSaveAOISaving = createAction('setSaveAOISaving');
+export const setSaveAOISaved = createAction('setSaveAOISaved');
+export const setSaveAOIDeleted = createAction('setSaveAOIDeleted');
+export const resetSaveAOI = createAction('resetSaveAOI');
+export const clearSaveAOIError = createAction('clearSaveAOIError');
+
+export const setSaveAOISettings = createThunkAction(
```

Comment:
I am wondering about this. What state are you saving to the url from the modal? Maybe we can take a look at this together.